lib/raidenclient/RaidenClient.ts

```diff
@@ -272,14 +278,18 @@ class RaidenClient extends SwapClient {
   }
 
   /**
-   * Returns the total balance available across all channels.
+   * Returns the total balance available across all channels for a specified currency.
    */
-  public channelBalance = async (): Promise<ChannelBalance> => {
-    // TODO: refine logic to determine balance per token rather than all combined
-    const channels = await this.getChannels();
+  public channelBalance = async (currency?: string): Promise<ChannelBalance> => {
+    if (!currency) {
+      return { balance: 0, pendingOpenBalance: 0 };
+    }
+
+    const channels = await this.getChannels(this.tokenAddresses.get(currency));
     const balance = channels.filter(channel => channel.state === 'opened')
       .map(channel => channel.balance)
-      .reduce((acc, sum) => sum + acc, 0);
+      .reduce((sum, acc) => sum + acc, 0)
+      / (RaidenClient.UNITS_PER_CURRENCY[currency] || 1);
```

Comment:
One last nit - when commenting out this line the tests are still passing. We might want to consider covering the conversion in `channelBalance calculates the total balance of open channels for a currency` test as well. Up to you :).

Reply:
Makes sense, I'll add this test case and merge.

Reply:
Actually, I remember now that I tried to do this originally but in its current state I'd have to mock a static value and I'm not really sure how to do that. Maybe revisit this when we get the units per currency from the database?

Reply:
Fair enough.